pkg/audit/manager.go

```diff
@@ -526,10 +526,12 @@ func (am *Manager) writeAuditResults(ctx context.Context, resourceList []schema.
 		}
 		if len(updateConstraints) > 0 {
 			if am.ucloop != nil {
-				close(am.ucloop.stop)
 				select {
 				case <-am.ucloop.stopped:
+					close(am.ucloop.stop)
```

Comment:
This will make the code pseudo-serial -- each goroutine will be blocked until the previous goroutine exits. The biggest problem with that behavior is that subsequent audit runs may be delayed by an inability to update resources.

Can we refactor `ucloop.update()` so that it updates ALL constraint GVKs (i.e. move the `for _, constraintGvk := range resourceList` into `ucloop.update()`)

That way, subsequent audit runs will still preempt reporting and there will be fewer goroutines, which should make the code easier to reason about?

This should retain the same performance as the current proposed solution.